mlflow/models/dependencies_schema.py

```diff
@@ -0,0 +1,162 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Dict, List, Optional
+
+from mlflow.models.resources import ResourceType
+from mlflow.utils.annotations import experimental
+
+DATABRICKS_VECTOR_SEARCH_PRIMARY_KEY = "__databricks_vector_search_primary_key__"
+DATABRICKS_VECTOR_SEARCH_TEXT_COLUMN = "__databricks_vector_search_text_column__"
+DATABRICKS_VECTOR_SEARCH_DOC_URI = "__databricks_vector_search_doc_uri__"
+DATABRICKS_VECTOR_SEARCH_OTHER_COLUMNS = "__databricks_vector_search_other_columns__"
+
+
+@experimental
+def set_vector_search_schema(
+    primary_key: str,
+    text_column: str,
+    doc_uri: Optional[str] = None,
+    other_columns: Optional[List[str]] = None,
+):
+    """
+    After defining your vector store in a Python file or notebook, call
+    set_vector_search_schema() so that, when MLflow retrieves documents during
+    model inference, MLflow can interpret the fields in each retrieved document and
+    determine which fields correspond to the document text, document URI, etc.
+
+    Args:
+        primary_key: The primary key of the vector index.
+        text_column: The name of the text column to use for the embeddings.
+        doc_uri: The name of the column that contains the document URI.
+        other_columns: A list of other columns that are part of the vector index
+                          that need to be retrieved during trace logging.
+        Note: Make sure the text column specified is in the index.
+
+        Example:
+
+        .. code-block:: python
+
+            from mlflow.models import set_vector_search_schema
+
+            set_vector_search_schema(
+                primary_key="chunk_id",
+                text_column="chunk_text",
+                doc_uri="doc_uri",
+                other_columns=["title"],
+            )
+    """
+    globals()[DATABRICKS_VECTOR_SEARCH_PRIMARY_KEY] = primary_key
+    globals()[DATABRICKS_VECTOR_SEARCH_TEXT_COLUMN] = text_column
+    globals()[DATABRICKS_VECTOR_SEARCH_DOC_URI] = doc_uri
+    globals()[DATABRICKS_VECTOR_SEARCH_OTHER_COLUMNS] = other_columns or []
+
+
+def _get_vector_search_schema():
+    """
+    Get the vector search schema defined by the user.
+
+    Returns:
+        VectorSearchIndex: The vector search index schema.
+    """
+    return VectorSearchIndexSchema(
+        name="vector_search_index",
+        primary_key=globals().get(DATABRICKS_VECTOR_SEARCH_PRIMARY_KEY),
+        text_column=globals().get(DATABRICKS_VECTOR_SEARCH_TEXT_COLUMN),
+        doc_uri=globals().get(DATABRICKS_VECTOR_SEARCH_DOC_URI),
+        other_columns=globals().get(DATABRICKS_VECTOR_SEARCH_OTHER_COLUMNS),
+    )
+
+
+class DependenciesType(Enum):
+    """
+    Enum to define the different types of dependencies of the model.
+    """
+
+    VECTOR_SEARCH_INDEX = ResourceType.VECTOR_SEARCH_INDEX.value
```

Comment:
Sorry, to clarify my previous comment: Can we make the `type` field on line 88 a `ResourceType` and remove `DependenciesType`?